tests/unit/reader/test_schema_registry.py

```diff
@@ -89,6 +91,8 @@ def test_put_schema(schema_api_response):
         status=200,
     )
 
-    schema_id = SchemaRegistry(TEST_SCHEMA_REGISTRY).put_schema("test", TEST_SCHEMA)
+    schema_id = SchemaRegistry(TEST_SCHEMA_REGISTRY).put_schema(
+        "test", json.dumps(TEST_SCHEMA)
```

Comment:
@farbodahm type checker was complaining about this (`put_schema` accepts str but received dict). This silences the type checker, but tbh the test passes in both cases. Should we change/review the types accepted by `put_schema` ?